homeassistant/components/ezviz/config_flow.py

```diff
@@ -41,70 +37,98 @@ class EzvizConfigFlow(ConfigFlow, domain=DOMAIN):
+            raise InvalidAuth from err
+
+        return self.async_create_entry(title=data[CONF_USERNAME], data=data)
+
     @staticmethod
     @callback
     def async_get_options_flow(config_entry):
         """Get the options flow for this handler."""
         return EzvizOptionsFlowHandler(config_entry)
 
-    async def async_step_import(
-        self, user_input: Optional[ConfigType] = None
-    ) -> Dict[str, Any]:
-        """Handle a flow initiated by configuration file."""
-        return await self.async_step_user(user_input)
-
-    async def async_step_user(
-        self, user_input: Optional[ConfigType] = None
-    ) -> Dict[str, Any]:
+    async def async_step_user(self, user_input=None):
         """Handle a flow initiated by the user."""
         if self._async_current_entries():
             return self.async_abort(reason="single_instance_allowed")
 
         errors = {}
-        default_username = ""
 
         if user_input is not None:
             if CONF_TIMEOUT not in user_input:
                 user_input[CONF_TIMEOUT] = DEFAULT_TIMEOUT
 
-            default_username = user_input[CONF_USERNAME]
-
             try:
-                await self.hass.async_add_executor_job(
-                    validate_input, self.hass, user_input
-                )
-            except PyEzvizError:
-                errors["base"] = "cannot_connect"
+                return await self._validate_and_create(user_input)
+
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+
+            except AbortFlow:
+                raise
+
             except Exception:  # pylint: disable=broad-except
                 _LOGGER.exception("Unexpected exception")
                 return self.async_abort(reason="unknown")
-            else:
-                return self.async_create_entry(
-                    title=user_input[CONF_USERNAME],
-                    data=user_input,
-                )
-
-        data_schema = {
-            vol.Required(CONF_USERNAME, default=default_username): str,
-            vol.Required(CONF_PASSWORD): str,
-            vol.Optional(CONF_REGION, default=DEFAULT_REGION): str,
-        }
 
         return self.async_show_form(
-            step_id="user",
-            data_schema=vol.Schema(data_schema),
-            errors=errors or {},
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors or {}
         )
 
+    async def async_step_import(self, import_config):
+        """Handle config import from yaml."""
+        try:
+            return await self._validate_and_create(import_config)
+
+        except InvalidAuth:
+            _LOGGER.error("Error importing Ezviz platform config: invalid auth.")
+            return self.async_abort(reason="invalid_auth")
+
+        except AbortFlow:
+            raise
+
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception(
+                "Error importing ezviz platform config: unexpected exception."
```

Comment:
We don't end logging messages with period.

Reply:
Removed period from config_flow logging entries.